src/pbr/pbr.cpp

```diff
@@ -0,0 +1,329 @@
+#include <iostream>
+#include <GL/glew.h>
+#include <glm/glm.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+#include "pbr.hpp"
+#include "plane.hpp"
+#include "cube.hpp"
+#include "textures.hpp"
+
+PBRApp::PBRApp(
+    const char* tWindowName,
+    const int tWindowWidth,
+    const int tWindowHeight
+    ) : CameraApp(tWindowName, tWindowWidth, tWindowHeight) {}
+
+void PBRApp::setUpScene() {
+  glEnable(GL_DEPTH_TEST);
+  glEnable(GL_TEXTURE_CUBE_MAP_SEAMLESS);
+  glDepthFunc(GL_LEQUAL);
+
+  glGenFramebuffers(1, &mCaptureFBO);
+  glGenRenderbuffers(1, &mCaptureRBO);
+
+  buildEnvCubemap();
+  buildIrradianceCubemap();
+  buildPrefilteredEnvMap();
+  buildBRDFLUT();
+
+  glDeleteFramebuffers(1, &mCaptureFBO);
+  glDeleteRenderbuffers(1, &mCaptureRBO);
+
+
+  //buildGeometry();
+
+  mPBRShader.use();
+  //mAlbedo = loadTexture("./materials/streaked_metal/albedo.png");
+  //mAlbedo = loadTexture("./materials/rock/albedo.png");
+  //mAlbedo = loadTexture("./materials/rusted_iron/albedo.png");
+  //mAlbedo = loadTexture("./materials/marble/albedo.png");
+  mAlbedo = loadTexture("./materials/plastic/albedo.png");
```

Comment:
we can extract these to separate functions instead (or single function that accepts name of the material)